provisioning/service/src/Manager/EnrollmentGroupManager.cs

```diff
@@ -81,24 +98,25 @@ internal static class EnrollmentGroupManager
         /// <see cref="ProvisioningServiceClient.DeleteEnrollmentGroupAsync(EnrollmentGroup)"/>
         ///
         /// <param name="enrollmentGroup">is an <see cref="EnrollmentGroup"/> that describes the enrollment that will be deleted. It cannot be <code>null</code>.</param>
-        /// <exception cref="ArgumentException">if the provided parameter is not correct.</exception>
+        /// <exception cref="ArgumentNullException">if the provided parameter is not correct.</exception>
         /// <exception cref="ProvisioningServiceClientTransportException">if the SDK failed to send the request to the Device Provisioning Service.</exception>
         /// <exception cref="ProvisioningServiceClientException">if the Device Provisioning Service was not able to execute the delete operation.</exception>
-        internal static Task DeleteAsync(
+        internal static async Task DeleteAsync(
             IContractApiHttp contractApiHttp,
             EnrollmentGroup enrollmentGroup,
             CancellationToken cancellationToken)
         {
-            /* SRS_ENROLLMENT_GROUP_MANAGER_28_010: [The DeleteAsync shall throw ArgumentException if the provided enrollmentGroup is null.] */
+            /* SRS_ENROLLMENT_GROUP_MANAGER_28_010: [The DeleteAsync shall throw ArgumentNullException if the provided enrollmentGroup is null.] */
             if (enrollmentGroup == null)
             {
-                throw new ArgumentException(nameof(enrollmentGroup));
+                throw new ArgumentNullException(nameof(enrollmentGroup));
             }
 
             /* SRS_ENROLLMENT_GROUP_MANAGER_28_011: [The DeleteAsync shall sent the Delete HTTP request to remove the enrollmentGroup.] */
-            return contractApiHttp.DeleteAsync(
+            await contractApiHttp.RequestAsync(
+                HttpMethod.Delete,
                 GetEnrollmentUri(enrollmentGroup.EnrollmentGroupId),
-                enrollmentGroup.ETag,
+                null,
                 null,
                 null,
                 cancellationToken);
```

Comment:
Here and on every await : we need to always specify .ConfigureAwait. For libraries that's usually set to false, to allow the task continuation to execute on any available thread.

Reply:
I won't add more comments about ConfigureAwait but this applies to the entire code base.

Reply:
done for all.